### PR TITLE
/mob/living/simple_animal is now frozen. Add a unit test to block all new simple_animal subtypes.

### PR DESCRIPTION
**Pull request by Mothblocks**
@tralezab has assured me that basic mobs are now intuitive enough to be copied and pasted from even by more novice coders. He is also actively working on a contribution guide on how to transition old simple mobs, and how to create new basic mobs.

With this, we have agreed to completely freezing simple_animal. All new "simple" mobs MUST be basic mobs. Ask maintainers, especially Armhulen, for help in conversion.

As simple_animals get removed, they will be removed from the list (to prevent a compile error). If a non-simple_animal is in the list, likely through a botched find and replace, the test will fail. You need an **extremely** good reason to add to this list. Remember that nobody intends to do the wrong thing, but we will never get rid of these if we make exceptions.

### CODE REVIEW

File: code/modules/unit_tests/simple_animal_freeze.dm [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+		/mob/living/simple_animal/mouse/cable_lover,
+		/mob/living/simple_animal/mouse/gray,
+		/mob/living/simple_animal/mouse/white,
+		/mob/living/simple_animal/parrot,
+		/mob/living/simple_animal/parrot/natural,
+		/mob/living/simple_animal/parrot/poly,
+		/mob/living/simple_animal/parrot/poly/ghost,
+		/mob/living/simple_animal/pet,
+		/mob/living/simple_animal/pet/cat,
+		/mob/living/simple_animal/pet/cat/_proc,
```
Comment (Mothblocks) [inline code/modules/unit_tests/simple_animal_freeze.dm:359]: not a fan of you one bit

**Review by ShizCalev — APPROVED** [review]

File: code/modules/unit_tests/simple_animal_freeze.dm (hunk 1 above)
Comment (CapybaraExtravagante) [reply code/modules/unit_tests/simple_animal_freeze.dm:359]: https://adw.org/about-us/resources/request-an-exorcism/

**Comment by tralezab** [comment]
Working on a renewed guide for creating AI, and a new one for converting simple to basic

File: code/modules/unit_tests/simple_animal_freeze.dm (hunk 1 above)
Comment (tralezab) [reply code/modules/unit_tests/simple_animal_freeze.dm:359]: jesus

**Review by tralezab — APPROVED** [review]
great test

**Pull request merged**